internal/bminventory/inventory_v2_handlers.go

```diff
@@ -392,3 +394,36 @@ func (b *bareMetalInventory) V2ListFeatureSupportLevels(ctx context.Context, par
 	payload := featuresupport.SupportLevelsList
 	return installer.NewV2ListFeatureSupportLevelsOK().WithPayload(payload)
 }
+
+func (b *bareMetalInventory) V2GetPresignedForClusterCredentials(ctx context.Context, params installer.V2GetPresignedForClusterCredentialsParams) middleware.Responder {
+	presigned, err := b.getPresignedForClusterCredentials(ctx, &params.ClusterID, params.FileName)
+	if err != nil {
+		return err
+	}
+	return installer.NewV2GetPresignedForClusterCredentialsOK().WithPayload(presigned)
+}
+
+func (b *bareMetalInventory) getPresignedForClusterCredentials(ctx context.Context, clusterId *strfmt.UUID, fileName string) (*models.Presigned, middleware.Responder) {
```

Comment:
I think this can be part of the same function, `V2GetPresignedForClusterCredentials`.

You'd only really need a separate function here if you're creating some "Internal" version to share with the controllers or it's being shared between multiple API endpoints. I don't think either of those are happening here.

Reply:
Done